recipes-core/disk-encryption/init

```diff
@@ -25,6 +30,34 @@ log() {
     echo "$1" > /var/volatile/system-api.fifo
 }
 
+find_target_disk() {
+    local target_path
+    
+    # First try to find disk by LUN
+    target_path=$(readlink -f "/dev/disk/by-path/"*"scsi-0:0:0:${TARGET_LUN}" 2>/dev/null || true)
+    
+    if [ -n "$target_path" ] && [ -b "$target_path" ]; then
+        log "Found target disk at ${target_path} using LUN ${TARGET_LUN}"
+        echo "$target_path"
+        return 0
+    fi
+    
+    # Fallback: look for the first data disk that's not the boot disk
+    local boot_disk
+    boot_disk=$(mount | grep " on / " | cut -d' ' -f1 | sed 's/[0-9]*$//')
```

Comment:
This returns `rootfs` when two disks mounted inside the VM. How do you compare it then to the `lsblk` output which returns, e.g., `/dev/sdb`?

Reply:
This fetches the boot disk as you said. and then uses it later in line 50 to avoid while searching for the attached disk that is not the boot disk and uses that instead.

Reply:
It is like a fallback solution in case the attached disk was not assigned a LUN number during deployment. 
We could also delete this part and return an error instead ofc but I thought we could test with it and see too. wdyt?